engines/grim/lua/lrestore.cpp

```diff
@@ -581,4 +581,23 @@ void lua_Restore(SaveGame *savedState) {
 	savedState->endSection();
 }
 
+void lua_PatchGrimSave() {
+	// Since ResidualVM 0.2.0, a ResidualVM/ScummVM specific patch was provided broken.
+	// We patch here the code to fix all saves containing this invalid code.
+	// cf. bug #13139 and #14987
+	// The patch here doesn't match exactly the fixed patch.
+	// We minimize here the number of bytes to patch with an equivalent result.
+	TProtoFunc *tempProtoFunc = (TProtoFunc *)rootproto.next;
+	while (tempProtoFunc) {
+		if ((tempProtoFunc->lineDefined == 77) &&
+			(strcmp(tempProtoFunc->fileName->str, "Scripts\\vd.lua") == 0) &&
+			(memcmp(tempProtoFunc->code + 210, "\x22\x29\x39\x03\x32\x22\x30\x79\x02", 9) == 0)) {
+			tempProtoFunc->code[211] = 0x33;
+			tempProtoFunc->code[218] = 0x03;
+			break;
```

Comment:
How does this work? Are the saved patched forever or is the patch only saved when saving again?
I'm thinking if the save version should be bumped after patch is applied?

Reply:
It's patched in memory.
Only saving the game again will make the patch permanent.
In this case, the version will be bumped.